lua/wire/clientsidehttp.lua

```diff
@@ -0,0 +1,366 @@
+    		if func() then
+    			timer.Remove(id)
+    		end
+    	end)
+    end
+
+    -- void function(number uid, string body_compressed, number length); Internal: Do not call.
+    local function writeHTTPBody(uid,body,length)
+    	local segments = math.ceil(length/clHTTP.segmentSize)
+
+    	-- reliable netmessages are always sent in order
+    	for i=1,segments do
+    		local segment = body:sub(clHTTP.segmentSize * (i-1) + 1,clHTTP.segmentSize * i)
+    		net.Start(clHTTP.netMsgID)
+    			net.WriteUInt(uid,8)						-- this is who we are
+    			net.WriteBool(false)						-- No; the request did not error
+    			net.WriteBool(true)							-- yes we are writing a body
+    			net.WriteUInt(#segment,clHTTP.segmentBits)	-- the body is this much long
+    			net.WriteData(segment,#segment)				-- and here is the body
+    		net.SendToServer()
+
+    		coroutine.yield()
+    	end
+    end
+
+    -- void function(number uid, number length, table headers, number code); Internal: Do not call.
+    local function writeMetadata(uid,length,headers,code)
+    	net.Start(clHTTP.netMsgID)
+    		net.WriteUInt(uid,8)						-- this is who we are
+    		net.WriteBool(false)						-- No; the request did not error
+    		net.WriteBool(false)						-- we are ready to finalise the message
+    		-- CAPITAL SIN: net.WriteTable is horribly
+    		-- unoptimised; but there is no other
+    		-- elegant solution to write in a variable
+    		-- table
+    		net.WriteTable(headers)						-- these are the headers
+    		net.WriteUInt(code,12)						-- and this is the code
+    	net.SendToServer()
+    end
+
+    -- void function(number uid, string err); Internal: Do not call.
+    local function returnFailure(uid,err)
+    	net.Start(clHTTP.netMsgID)
+    		net.WriteUInt(uid,8)	-- this is who we are
+    		net.WriteBool(true)		-- unfortunately, the request caused an error
+    		net.WriteString(err)	-- and this is the code
+    	net.SendToServer()
+    end
+
+    -- void function(number uid, string body, number length, table headers, number code); Internal: Do not call.
+    local function returnData(uid,body,length,headers,code)
+    	local body_compressed = util.Compress(body) or ""
+
+    	writeHTTPBody(uid,body_compressed,#body_compressed)
+    	writeMetadata(uid,length,headers,code)
+
+    	return true
+    end
+
+    -- void function(string url, number uid); Internal: Do not call.
+    local function performRequest(url,uid)
+    	clHTTP.rawRequest(url,function(body,length,headers,code)
+    		timerCoroutine("wire_e2_cl_http_"..uid,clHTTP.sendInterval,function()
+    			return returnData(uid,body,length,headers,code)
+    		end)
+    	end,function(err)
+    		returnFailure(uid,err)
+    	end)
+    end
+
+    -- void function(void); Internal: Do not call.
+    local function handleIncomingRequest()
+    	local url,uid = net.ReadString(),net.ReadUInt(8)
+
+    	performRequest(url,uid)
+    end
+
+    net.Receive(clHTTP.netMsgID,function(_,ply)
```

Comment:
net.Receive doesn't callback with a player argument on client, and why does it call "handleIncomingRequest" with said player argument when it doesn't take one?